test/runtime/benchmark.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package RuntimeTest
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/logging"
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("BenchmarkNetperfPerformance", func() {
+	var (
+		vm          *helpers.SSHMeta
+		monitorStop = func() error { return nil }
+
+		log    = logging.DefaultLogger
+		logger = logrus.NewEntry(log)
+
+		PerfLogFile   = "l4bench_perf.log"
+		PerfLogWriter bytes.Buffer
+	)
+
+	BeforeAll(func() {
+		vm = helpers.InitRuntimeHelper(helpers.Runtime, logger)
+		ExpectCiliumReady(vm)
+	})
+
+	JustBeforeEach(func() {
+		monitorStop = vm.MonitorStart()
+	})
+
+	JustAfterEach(func() {
+		vm.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+		Expect(monitorStop()).To(BeNil(), "cannot stop monitor command")
+	})
+
+	AfterFailed(func() {
+		vm.ReportFailed(
+			"cilium service list",
+			"cilium policy get")
+	})
+
+	AfterEach(func() {
+		LogPerm := os.FileMode(0666)
+		testPath, err := helpers.CreateReportDirectory()
+		Expect(err).Should(BeNil(), "cannot create log file")
+		helpers.WriteOrAppendToFile(filepath.Join(testPath, PerfLogFile), PerfLogWriter.Bytes(), LogPerm)
+		PerfLogWriter.Reset()
+	}, 500)
+
+	createContainers := func() {
+		By("create Client container")
+		vm.ContainerCreate(helpers.Client, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.client")
+		By("create Server containers")
+		vm.ContainerCreate(helpers.Server, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.server")
+		vm.PolicyDelAll()
+		vm.WaitEndpointsReady()
```

Comment:
`Expect( 		vm.WaitEndpointsReady() ).To(BeNil(), "Endpoints are not ready")`

Reply:
updated to suggested Expect(...)